packages/lsp-extension/src/index.ts

```diff
@@ -302,12 +307,24 @@ function addRunningSessionManager(
     )
   });
 }
+
+const adapterTrackerPlugin: JupyterFrontEndPlugin<IWidgetLSPAdapterTracker> = {
+  id: '@jupyterlab/lsp-extension:tracker',
+  description: 'Provides the `WidgetLSPAdapter`s tracker.',
+  autoStart: true,
+  provides: IWidgetLSPAdapterTracker,
+  activate: (app: JupyterFrontEnd<LabShell>): IWidgetLSPAdapterTracker => {
```

Comment:
This seems related to the use of the `currentChanged` signal here:

https://github.com/jupyterlab/jupyterlab/blob/0b33a5ca9318d909233914adb0cbd90d3bc7a472/packages/lsp/src/adapters/tracker.ts#L28-L45

Reply:
Ideally the tracker plugin could also work with shells that do not implement a `currentChanged` signal, or for which the signature is different.

In this case the plugin could check `currentWidget`, which is part of the `JupyterFrontEnd.IShell` interface.

Reply:
Yes, I brought it up earlier in https://github.com/jupyterlab/jupyterlab/pull/14920#discussion_r1316507818.

The problem with `IShell` is that it indeed is missing a few things which are really handy (there is a reason why these were added to `LabShell`)
- https://github.com/jupyterlab/jupyterlab/issues/14782
- https://github.com/jupyterlab/jupyterlab/issues/14883

Is the Notebook compatibility problem related to one of the above, or a separate issue due to divergence of shells? Would it be a good idea to make the Notebook shell conform closer it Lab shell even if sometimes the signals/attributes would be no-op for Notebook?

Reply:
> Would it be a good idea to make the Notebook shell conform closer it Lab shell even if sometimes the signals/attributes would be no-op for Notebook?

Yes I think we can do that. Currently the signature is as follows: https://github.com/jupyter/notebook/blob/c3ec39e8d7428aac2cda6b0a9fed17643eab5210/packages/application/src/shell.ts#L166

```ts
get currentChanged(): ISignal<NotebookShell, void>
```

While in JupyterLab:

https://github.com/jupyterlab/jupyterlab/blob/0b33a5ca9318d909233914adb0cbd90d3bc7a472/packages/application/src/shell.ts#L545

Which is why it is causing an issue in Notebook 7 since there is no `newValue`. But I guess `INotebookShell` could align its signature with the one from JupyterLab to reduce friction for that particular case. But this wouldn't prevent the LSP plugin to use other `ILabShell` specific APIs if this is what it expects.